src/home/homectl.c

```diff
@@ -0,0 +1,3607 @@
+                                        ret = -EINVAL;
+
+                                continue;
+                        }
+
+                        r = sd_bus_call_method(
+                                        bus,
+                                        "org.freedesktop.home1",
+                                        "/org/freedesktop/home1",
+                                        "org.freedesktop.home1.Manager",
+                                        "GetUserRecordByName",
+                                        &error,
+                                        &reply,
+                                        "s",
+                                        *i);
+                } else {
+                        r = sd_bus_call_method(
+                                        bus,
+                                        "org.freedesktop.home1",
+                                        "/org/freedesktop/home1",
+                                        "org.freedesktop.home1.Manager",
+                                        "GetUserRecordByUID",
+                                        &error,
+                                        &reply,
+                                        "u",
+                                        (uint32_t) uid);
+                }
+
+                if (r < 0) {
+                        log_error_errno(r, "Failed to inspect home: %s", bus_error_message(&error, r));
+                        if (ret == EXIT_SUCCESS)
+                                ret = r;
+
+                        continue;
+                }
+
+                r = sd_bus_message_read(reply, "sbo", &json, &incomplete, NULL);
+                if (r < 0) {
+                        bus_log_parse_error(r);
+                        if (ret == EXIT_SUCCESS)
+                                ret = r;
+
+                        continue;
+                }
+
+                r = json_parse(json, JSON_PARSE_SENSITIVE, &v, NULL, NULL);
+                if (r < 0) {
+                        log_error_errno(r, "Failed to parse JSON identity: %m");
+                        if (ret == EXIT_SUCCESS)
+                                ret = r;
+
+                        continue;
+                }
+
+                hr = user_record_new();
+                if (!hr)
+                        return log_oom();
+
+                r = user_record_load(hr, v, USER_RECORD_LOAD_REFUSE_SECRET|USER_RECORD_LOG);
+                if (r < 0) {
+                        if (ret == EXIT_SUCCESS)
+                                ret = r;
+
+                        continue;
+                }
+
+                hr->incomplete = incomplete;
+                dump_home_record(hr);
+        }
+
+        return ret;
+}
+
+static int authenticate_home(int argc, char *argv[], void *userdata) {
+        _cleanup_(sd_bus_flush_close_unrefp) sd_bus *bus = NULL;
+        _cleanup_(strv_freep) char **mangled_list = NULL;
+        int r, ret = EXIT_SUCCESS;
```

Comment:
Using EXIT_SUCCESS is confusing. EXIT_* codes are non-negative, but afaict, `ret` is used for negative failure values.